core/primitives/src/validator_mandates/compute_price.rs

```diff
@@ -0,0 +1,246 @@
+use {
+    super::ValidatorMandatesConfig,
+    near_primitives_core::types::Balance,
+    std::cmp::{min, Ordering},
+};
+
+/// Given the stakes for the validators and the target number of mandates to have,
```

Comment:
I don't fully understand how complexity of `compute_mandate_price` is justified. I want to discuss it, as it is going to become part of protocol.

Is there a way to make it simpler by sacrificing some accuracy? I think it is completely fine if formula is simpler but number of mandates is higher.
Is there a way to make return value more consistent? I am not comfortable with 25 iterations. If stake changes by 1 unit or some small value, what guarantees that resulting price doesn't change significantly? The current function already leads to some magic constants in results. If some day we observe unexpected number of mandates on mainnet, it could be very problematic to find what went wrong and how to fix the number.

If we proceed with current approach, please comment more why the iterative approach works. I assume this is due to fixed point theorem but it's not clear why `(S - \sum_i r_i) / N` is a contraction, especially when stakes are low (right?).